server/src/repositories/config.repository.ts

```diff
@@ -102,6 +102,11 @@ export interface EnvData {
 
   workers: ImmichWorker[];
 
+  plugins: Array<{
+    enabled: boolean;
+    manifestPath: string;
+  }>;
+
```

Comment:
This is not correct. The config repository loads configuration information. Specifically it loads it from environment variables. The type of information should be something like this:
```
IMMICH_PLUGINS_ENABLED=true
IMMICH_PLUGINS_INSTALL_FOLDER=/path/to/plugins
```

```ts
{
  plugins: {
    enabled: boolean;
    installFolder: string;
   }
 }
 ```

This is only used for stuff the user should be able to configure. We're going to package the core plugin into the image, so it should just be available at an _unconfigurable_ path. I would say we actually just add `resourcePaths.corePlugin` and set it to `join(buildFolder, 'corePlugin')`.